core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -420,4 +425,113 @@ protected WriteReplicaResult shardOperationOnReplica(BulkShardRequest request, I
         return next;
     }
 
+    public <Request extends ReplicatedWriteRequest<Request>, Response extends ReplicationResponse & WriteResponse>
+    WritePrimaryResult<Request, Response> executeSingleItemBulkRequestOnPrimary(
+            Request request, IndexShard primary) throws Exception {
+        BulkItemRequest[] itemRequests = new BulkItemRequest[1];
+        WriteRequest.RefreshPolicy refreshPolicy = request.getRefreshPolicy();
+        request.setRefreshPolicy(WriteRequest.RefreshPolicy.NONE);
```

Comment:
thanks for pointing it out. This was left over from using the bulk action directly, I removed mutating the request